lib/teleterm/apiserver/handler/handler_gateways.go

```diff
@@ -74,8 +77,8 @@ func (s *Handler) RemoveGateway(ctx context.Context, req *api.RemoveGatewayReque
 	return &api.EmptyResponse{}, nil
 }
 
-func newAPIGateway(gateway gateway.Gateway) (*api.Gateway, error) {
-	command, err := gateway.CLICommand()
+func (s *Handler) newAPIGateway(gateway gateway.Gateway) (*api.Gateway, error) {
+	command, err := s.DaemonService.GetGatewayCLICommand(gateway)
```

Comment:
It looks quite strange that `s.DaemonService` during `GetGatewayCLICommand`  taks the `gateway` dependency and injects the result to api.Gateway object. 
Maybe we extract a free function from this call to show obj dependency.

Reply:
what do you mean? Something like this?
```
	for _, gw := range gws {
		command, err := s.DaemonService.GetGatewayCLICommand(gw)
		if err != nil {
			return nil, trace.Wrap(err)
		}

		apiGateway, err := newAPIGateway(gw, makeGatewayCLICommand(command))
		if err != nil {
			return nil, trace.Wrap(err)
		}

		apiGws = append(apiGws, apiGateway)
	}

Reply:
I was thinking of something like `CreateGatewaycommand(storage, gateway)`  but it looks like you cannot get the Storage from DemonService so this suggestion is  irrelevant.

Reply:
Maybe we can pass in gateway URI instead of gateway here and let the daemon do a little more work. Or just leave the current implementation (without this refactoring) if that looks cleaner at the API level.

Reply:
For me this form makes hard to track dependency between objects.